debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -364,21 +365,58 @@ public Lsn getFromLsn() {
     }
 
     public Set<SqlServerChangeTable> listOfChangeTables(String databaseName) throws SQLException {
-        return queryAndMap(replaceDatabaseNamePlaceholder(GET_LIST_OF_CDC_ENABLED_TABLES, databaseName), rs -> {
+        return listOfChangeTables(databaseName, Lsn.NULL, Lsn.NULL);
+    }
+
+    public Set<SqlServerChangeTable> listOfChangeTables(String databaseName, Lsn fromLsn, Lsn toLsn) throws SQLException {
+        Map<Integer, List<String>> columns = queryAndMap(
+                replaceDatabaseNamePlaceholder(GET_LIST_OF_CDC_ENABLED_COLUMNS, databaseName),
+                rs -> {
+                    Map<Integer, List<String>> result = new HashMap<>();
+                    while (rs.next()) {
+                        int changeTableObjectId = rs.getInt(1);
+                        if (!result.containsKey(changeTableObjectId)) {
+                            result.put(changeTableObjectId, new LinkedList<>());
+                        }
+
+                        result.get(changeTableObjectId).add(rs.getString(3));
+                    }
+                    return result;
+                });
+
+        final ResultSetMapper<Set<SqlServerChangeTable>> mapper = rs -> {
             final Set<SqlServerChangeTable> changeTables = new HashSet<>();
             while (rs.next()) {
+                int changeTableObjectId = rs.getInt(4);
                 changeTables.add(
                         new SqlServerChangeTable(
                                 new TableId(databaseName, rs.getString(1), rs.getString(2)),
                                 rs.getString(3),
-                                rs.getInt(4),
-                                Lsn.valueOf(rs.getBytes(6)),
-                                Lsn.valueOf(rs.getBytes(7)),
-                                Arrays.asList(BRACKET_PATTERN.matcher(Optional.ofNullable(rs.getString(15)).orElse(""))
-                                        .replaceAll("").split(", "))));
+                                changeTableObjectId,
+                                Lsn.valueOf(rs.getBytes(5)),
+                                columns.get(changeTableObjectId)));
             }
             return changeTables;
-        });
+        };
+
+        String query = replaceDatabaseNamePlaceholder(GET_LIST_OF_CDC_ENABLED_TABLES, databaseName);
+
+        if (toLsn.isAvailable()) {
+            return prepareQueryAndMap(query.replace(STATEMENTS_PLACEHOLDER, " WHERE ct.start_lsn <= ?"),
+                    ps -> {
+                        ps.setBytes(1, toLsn.getBinary());
+                        ps.setString(2, databaseName);
+                        ps.setString(3, databaseName);
+                    },
+                    mapper);
+        }
+
```

Comment:
```suggestion
        }
        else {
```

Let's make the flow a bit more apparent.

Reply:
I was following the "no else after return/throw/etc" rule which often helps reduce the code indentation and is part of several coding standards. Fixed.

Reply:
Ok, thx. Indeed I always avoid that style, but as you say, it's a matter of taste and we should manifest that somewhere. Thanks for updating it.